### PR TITLE
SAK-50097 Dashboard webapi initialization replace nulls with empty lists

### PR DESCRIPTION
**Pull request by jkjanetschek**

### CODE REVIEW

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java [hunk 1]
```diff
@@ -122,13 +122,13 @@ public void init() {
         boolean tasksEnabled = serverConfigurationService.getBoolean(PortalConstants.PROP_DASHBOARD_TASKS_ENABLED, false);
 
         // Load up all the available widgets, from properties
-        courseWidgets = serverConfigurationService.getStringList("dashboard.course.widgets", null);
+        courseWidgets = new ArrayList<>(serverConfigurationService.getStringList("dashboard.course.widgets", null));
```
Comment (ern) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java:125]: I think it makes more sense to use the default with emptyList()?
```suggestion
        courseWidgets = serverConfigurationService.getStringList("dashboard.course.widgets", Collections.emptyList());
```

Comment (jkjanetschek) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java:125]: I don't quite understand what you mean. My understanding is that the default is only used if the serverConfigurationService does not find values for a given property. When the argument for the default value is null, a new ArrayList<>() is returned.
However, if the serverConfigurationService does find values for this property, Arrays.asList(values) is returned and the subsequent method remove fails.

Therefore, setting dflt to Collections.emptyList() does not fix the issue of this JIRA.

Comment (ottenhoff) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java:125]: I agree with your read of the code

**Review by ottenhoff — APPROVED** [review]

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java (hunk 1 above)
Comment (ern) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java:125]: My apologies @jkjanetschek I didn't realize that `getStringList` when null is the default an empty list is returned.

however looking at that method it doesn't appear that a null should be returned. So not sure why the wrapping the result in another list is needed?

**Pull request merged**